lib/awful/keygrabber.lua

```diff
@@ -27,11 +101,99 @@ local function grabber(mod, key, event)
     end
 end
 
+local function runner(self, modifiers, key, event)
+    -- Stop the keygrabber with the `release_key`
+    if (key == self.release_key or conversion[key] == self.release_key)
+        and event == self.release_event and self.release_key then
+        self:stop(key, modifiers)
+        return false
+    end
+
+    -- Stop when only a subset of keys are allowed and it isn't one of them.
+    if self._private.allowed_keys and not self._private.allowed_keys[key] then
+        self:stop(key, modifiers)
+        return false
+    end
+
+    -- Support multiple release keys
+    if type(self.release_key) == "table" then
+        for _, k in ipairs(self.release_key) do
+            if (k == key or conversion[key] == k) and event == self.release_event then
+                self:stop(k, modifiers)
+                return false
+            end
+        end
+    end
+
+    local is_modifier = conversion[key] ~= nil
+
+    -- Reset the inactivity timer on each events.
+    if self._private.timer and self._private.timer.started then
+        self._private.timer:again()
+    end
+
+    -- Record the key sequence
+    if key == "BackSpace" and self.sequence:len() > 0 then
+        self.sequence = self.sequence:sub(
+            1, self.sequence:len() - 1
+        )
```

Comment:
This breaks with UTF8. I vaguely remember that we have ugly code in `awful.prompt` that handles this via some magic, but I cannot find it right now... :-/

So, no idea what to do about this.

Edit: I just tested this, and, well, I need to press backspace twice in a prompt to get rid of an `ä` again.